framework/src/application/node/forger/strategies.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { MinHeap, MaxHeap } = require('@liskhq/lisk-transaction-pool');
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
+
+class HighFeeForgingStrategy {
+	constructor({
+		// components
+		logger,
+		// Modules
+		chainModule,
+		transactionPoolModule,
+		// constants
+		maxPayloadLength,
+	}) {
+		this.chainModule = chainModule;
+		this.transactionPoolModule = transactionPoolModule;
+		this.logger = logger;
+
+		this.constants = { maxPayloadLength: BigInt(maxPayloadLength) };
+	}
+
+	async getTransactionsForBlock() {
+		// Initialize array to select transactions
+		const readyTransactions = [];
+
+		// Initialize block size with 0
+		let blockPayloadSize = BigInt(0);
+
+		// Initialize state store which will be discarded after selection
+		const stateStore = this.chainModule.newStateStore();
+
+		// Get processable transactions from transaction pool
+		const processableTransactions = this.transactionPoolModule.getProcessableTransactions();
+
+		// Convert sender transactions to min heap by nonce
+		const transactionsBySender = Object.keys(processableTransactions).reduce(
+			(obj, senderId) => {
+				const minHeapByNonce = new MinHeap();
+				processableTransactions[senderId].forEach(t =>
+					minHeapByNonce.push(t.nonce, t),
+				);
+				obj[senderId] = minHeapByNonce;
+
+				return obj;
+			},
+			{},
+		);
+
+		// Loop till we have last account exhausted to pick transactions
+		while (Object.keys(transactionsBySender).length !== 0) {
+			const feePriorityHeap = new MaxHeap();
```

Comment:
this should be out side of the loop so that you don't have to recreate the heap every time

Reply:
We have to re-create this heap every time, as the lowest nonce transaction per account can change in each iteration. This heap actually contains once lowest transaction per account. We can use an array with sort, but that's another point of not using heap.

Reply:
yes, but when you choose the transaction from the account, you can just pick the next one from the same account and insert it into the heap?

Reply:
Yes, we can do it but to manage different scenarios, when to re-create the heap and when to put the same transaction in existing heap will make it more complex to maintain.

Reply:
from my understanding, you never need to recreate the heap.
My understanding what need to happen is this, isn't it?
```
1. create max heap
2. put the first transaction from each account
3. peek max heap
4. try apply if it's ok, pop and add next transaction from the same account. if invalid, pop and skip inserting all transactions from the same account
5. repeat 3
```